lib/typescript-extensions.js

```diff
@@ -0,0 +1,169 @@
+        '@typescript-eslint/func-call-spacing': ['error', 'never'],
+
+        // Defined by NI
+        indent: 'off',
+        '@typescript-eslint/indent': ['error', 4],
+
+        // Defined by Airbnb
+        'init-declarations': 'off',
+        '@typescript-eslint/init-declarations': 'off',
+
+        // Defined by Airbnb
+        'keyword-spacing': 'off',
+        '@typescript-eslint/keyword-spacing': ['error', {
+            before: true,
+            after: true,
+            overrides: {
+                return: { after: true },
+                throw: { after: true },
+                case: { after: true }
+            }
+        }],
+
+        // Defined by Airbnb
+        'lines-between-class-members': 'off',
+        '@typescript-eslint/lines-between-class-members': ['error', 'always', { exceptAfterSingleLine: false }],
+
+        // Defined by Airbnb
+        'no-array-constructor': 'off',
+        '@typescript-eslint/no-array-constructor': 'error',
+
+        // Defined by Airbnb
+        'no-dupe-class-members': 'off',
+        '@typescript-eslint/no-dupe-class-members': 'error',
+
+        // Defined by Airbnb
+        'no-duplicate-imports': 'off',
+        '@typescript-eslint/no-duplicate-imports': 'off',
+
+        // Defined by Airbnb
+        'no-empty-function': 'off',
+        '@typescript-eslint/no-empty-function': ['error', {
+            allow: [
+                'arrowFunctions',
+                'functions',
+                'methods'
+            ]
+        }],
+
+        // Defined by Airbnb
+        'no-extra-parens': 'off',
+        '@typescript-eslint/no-extra-parens': ['off', 'all', {
+            conditionalAssign: true,
+            nestedBinaryExpressions: false,
+            returnAssign: false,
+            ignoreJSX: 'all',
+            enforceForArrowConditionals: false,
+        }],
+
+        // Defined by Airbnb
+        'no-extra-semi': 'off',
+        '@typescript-eslint/no-extra-semi': 'error',
+
+        // Defined by Airbnb
+        'no-invalid-this': 'off',
+        '@typescript-eslint/no-invalid-this': 'off',
+
+        // Defined by NI
+        'no-loop-func': 'off',
+        '@typescript-eslint/no-loop-func': 'error',
+
+        // Defined by Airbnb
+        'no-loss-of-precision': 'off',
+        '@typescript-eslint/no-loss-of-precision': 'off',
+
+        // Defined by Airbnb
+        'no-magic-numbers': 'off',
+        '@typescript-eslint/no-magic-numbers': ['off', {
+            ignore: [],
+            ignoreArrayIndexes: true,
+            enforceConst: true,
+            detectObjects: false,
+        }],
+
+        // Defined by Airbnb
+        'no-redeclare': 'off',
+        '@typescript-eslint/no-redeclare': 'error',
+
+        // Defined by Airbnb
+        'no-shadow': 'off',
+        '@typescript-eslint/no-shadow': 'error',
+
+        // Defined by Airbnb
+        'no-unused-expressions': 'off',
+        '@typescript-eslint/no-unused-expressions': ['error', {
```

Comment:
**@typescript-eslint/no-unused-expressions**
[This rule](https://eslint.org/docs/rules/no-unused-expressions) flagged some Web Portal code which looked fine to me:
```ts
    ngOnDestroy(): void {
        this.selectionChangedSubscription && this.selectionChangedSubscription.unsubscribe();
   }
```
It's counting the method body as an "unused expression" but I like the short circuit function call syntax.

**Proposal**: set `allowShortCircuit` and `allowTernary` to `true.

Reply:
I think this would be better written using [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) as:
```js
this.selectionChangedSubscription?.unsubscribe();
```

There was also [previous discussion on this property in the js rules](https://ni.visualstudio.com/Users/_git/abarreto-misc/pullrequest/83582?_a=files&path=%2Fairbnb-js-review%2Frules%2Fbest-practices.js) where we decided to keep this configuration off.

Reply:
@rajsite 🚀